[layer-style]

stacyk opened the pull request:
![](https://source.unsplash.com/featured/?cute,animal&Fdafd)

I tested this in Opera 84 in order to view this in a browser without support for layers.

I am using the local bootstrap version of the css file but I kept the other commented out for now (probably could remove it at least until the known bug is fixed?)

![before](https://user-images.githubusercontent.com/1581694/187278945-a248d16a-ff65-459b-9cc3-418349f1fcad.jpg)

![after](https://user-images.githubusercontent.com/1581694/187278942-98f98432-6771-4625-bd71-7552f1ccd161.jpg)

I wanted to keep this as simple as possible, using only CSS, to show the polyfill in action but I do wonder if we should use a little javascript to edit the message and what is shown even if your browser supports layers... maybe that isn't necessary? I know @mirisuzanne made a handy CodePen that adds a class if layers is supported or not.

stacyk committed:
add webpack to devDependencies

stacyk committed:
smaller local bootstrap

stacyk commented:
![Screen Shot 2022-08-04 at 10 51 48 AM](https://user-images.githubusercontent.com/1581694/182891495-36b6e84f-aea0-4aba-8a4d-1a0f4592d7a7.jpg)

jgerigmeyer committed:
Merge branch 'main' into layer-style

* main:
  Fix build

jgerigmeyer committed:
lint

stacyk committed:
update markup

stacyk committed:
import to framework layer troubleshoot

stacyk committed:
remove unused Bootstrap for demo

stacyk committed:
style updates

stacyk committed:
udpate styles

stacyk committed:
clean up css and markup for demo

stacyk committed:
update spacing and layout for all screen sizes

stacyk commented:
![](https://github.trello.services/images/mini-trello-icon.png) [I want a demo project using the Layers polyfill](https://trello.com/c/WsV2XlJ5/17-i-want-a-demo-project-using-the-layers-polyfill)

mirisuzanne commented on README.md:
```diff
@@ -1,3 +1,4 @@
 
-Run `npx webpack` to build the project after clonining. Navigate to the `index.html` of the project in your browser of choice to view.
+Run `npx webpack` to build the project after cloning. Navigate to the
+`index.html` of the project in your browser of choice to view.
```
```suggestion
`src/index.html` of the project in your browser of choice to view.
```

mirisuzanne commented on src/index.html:
```diff
@@ -30,8 +46,8 @@ <h2>Change the CSS stylesheet below!</h2>
           <input type="radio" id="polyfilledVersion" name="stylesheet" value="polyfilled">
           <label for="polyfilled">Polyfilled</label>
```
```suggestion
          <input type="radio" id="polyfilled" name="stylesheet" value="polyfilled">
          <label for="polyfilled">Polyfilled</label>
```

Or the other way around. These currently don't match.

mirisuzanne reviewed (changes requested):
Looks good. Noted a clarification and an html bug. I think my main question is:

- this repo seems to be doing more than just the example, but I'm not sure what? Are we making this a repo just for the demo? If so, can we simplify and clarify? If not, can we clarify what it is for?
- I added a netlify build at [layers-polyfill-example.netlify.app](https://layers-polyfill-example.netlify.app/). Not sure if that's what we plan to share?
- I don't think this includes a step to test for support, and then load either the original or polyfilled version, depending on support? That might be a good thing to include in a demo.

stacyk committed:
fix input and label match, remove CDN import

stacyk commented:
> Looks good. Noted a clarification and an html bug.

@mirisuzanne I just pushed a fix for the label input mismatch.

> I think my main question is:
> 
> * this repo seems to be doing more than just the example, but I'm not sure what? Are we making this a repo just for the demo? If so, can we simplify and clarify? If not, can we clarify what it is for?

@mirisuzanne I went back and forth between showing (visually) what the site would look like before the polyfilled css is loaded, but it only shows if your browser doesn't support layers to begin with. And then thought about how this demo I THINK is supposed to show how the polyfill works, not necessarily how layers works. After considering both of those use cases, this is what I landed on, somewhere between showing layers and showing the polyfill in action. 

This demo is a bit tricky to see what is taking place since most browsers support layers and you wouldn't really notice a difference between the supported browsers and if the polyfill is being used. 

> * I don't think this includes a step to test for support, and then load either the original or polyfilled version, depending on support? That might be a good thing to include in a demo.

Perhaps we can check for support and then only show the form that asks a person to select the polyfilled version so they can see a clear difference between the non polyfilled and polyfilled versions. I wonder though, if you are on a supported browser and still wanted to see how the polyfill works (especially while we are still working out some kinks) that we may want to show the form for everyone regardless of whether their browser supports layers or not. 

Maybe we can show a link the the CSS file being used so they can inspect it easier? 

Currently, the top row shows a message that only shows two options: 
1. Layers isn't supported
2. Layers either is supported or the polyfilled CSS file is being used

If we test for support first, we can split this last one up to have these three:
1. Layers isn't supported, try the polyfill...
2. Layers isn't supported, the polyfilled CSS file is being used
3. Layers is supported, no polyfill needed.

mirisuzanne commented:
@stacyk sorry for the confusion, that's not exactly how I intended the first question. Let me try and clarify:

> this repo seems to be doing more than just the example, but I'm not sure what? Are we making this a repo just for the demo? If so, can we simplify and clarify? If not, can we clarify what it is for? 

The demo itself makes sense to me, and I think you made the right call on what to highlight. I'm curious about the purpose of _this github repository_. Does it only exist for the demo> If so, can we clarify that in the readme/name and also the code? Right now, it looks like this is the repo for the polyfill, but it's not. And it looks like there's code unrelated to the demo?

Does that make sense?

-----

For the second part, I like your proposal. The main goal for me is not that we need to show some difference between the two - but that this demo is meant to teach people how the polyfill is used, and best practice is generally to _only load polyfilled code when the browser doesn't have built in support_. I just want to make sure that we're teaching people how they can do that test-and-swap on their sites.

oluoluoxenfree commented:
@mirisuzanne I think i put the issues/questions here as otherwise i guess they would all live within the postcss repo? But i agree it is confusing and probably shouldn't be here. I think it'd be good to rename everything to reflect it's a demo

stacyk committed:
Update README.md

stacyk committed:
scannable readme with resources

stacyk committed:
lint

stacyk committed:
lint

stacyk committed:
add new line

stacyk commented on README.md:
```diff
@@ -1,3 +1,20 @@
-# cascade-layers-polyfill
+# cascade-layers-polyfill-demo
 
```
@mirisuzanne @sanajaved7 @oluoluoxenfree I updated the name here and in the repo project settings

stacyk commented on README.md:
```diff
@@ -1,3 +1,20 @@
+# cascade-layers-polyfill-demo
 
-Run `npx webpack` to build the project after clonining. Navigate to the `index.html` of the project in your browser of choice to view.
+This is a demo showing how the Cascade Layers Polyfill is used.
```
@mirisuzanne @sanajaved7 @oluoluoxenfree  too vague?

mirisuzanne replied:
maybe just add links to the layers article, and the polyfill? Oh, I see you have those later, which is also fine. But they could work here.

stacyk commented on README.md:
```diff
@@ -1,3 +1,20 @@
+This is a demo showing how the Cascade Layers Polyfill is used.
+
+
+## Local Testing
+
+To run this project locally, run `npx webpack` to build the project.
+Open `src/index.html` in your web browser.
+
```
@mirisuzanne @sanajaved7 @oluoluoxenfree I think we still need webpack if someone were to clone this to work on it locally?

oluoluoxenfree replied:
I think webpack is needed for the polyfill to work on this demo, as we're using a node module for the post css plugin.Will put it back, sorry for the confusion

stacyk reviewed (commented):
@mirisuzanne @sanajaved7 @oluoluoxenfree I am not sure why or what has changed, but when I run this locally now, and open the index.html in Opera 84 (the browser/version that doesn't support CSS Layers) I get an error when I select "Polyfilled" on the page. Am I missing a step? It didn't serve the polyfilled CSS file. It also happens in Chrome (and I assume elsewhere).  
![Screen Shot 2023-02-23 at 12 36 19 PM](https://user-images.githubusercontent.com/1581694/221002553-9599eff7-f1ec-4fe3-803e-7744ee0092ce.jpg)

mirisuzanne reviewed (commented):
Yeah, this is looking great. At this point I just have the questions you already noted.

oluoluoxenfree committed:
fix for not app;lying polyfilled stylesheet

oluoluoxenfree commented:
@stacyk I think I fixed the bug you mentioned but @sanajaved7 I'm not sure why if you double click on `polyfilled` accidental it deletes the other radio button??

sanajaved7 committed:
Merge branch 'main' into layer-style

sanajaved7 committed:
Fixing duplicate ID

jgerigmeyer committed:
Merge branch 'main' into layer-style

* main:
  Bump yaml and postcss-loader

jgerigmeyer committed:
upgrade deps

jgerigmeyer committed:
lint

jgerigmeyer committed:
Fix polyfill toggle

jgerigmeyer committed:
move index.html

jgerigmeyer committed:
use webpack production mode

jgerigmeyer committed:
support older browsers

jgerigmeyer committed:
add polyfill for nodelist.foreach

jgerigmeyer committed:
Check for layer support

jgerigmeyer committed:
Add instructions for checking for @layer support

jgerigmeyer commented:
@stacyk @mirisuzanne I added a check for `@layer` support, updating the banner accordingly. I also added some sample JS for how users can do this. Happy for feedback/cleanup on any of this.

https://deploy-preview-3--layers-polyfill-example.netlify.app/

@stacyk We might want to consider moving the sample buttons over with the radios, now that they're pushed down the page?

mirisuzanne commented on src/index.css:
```diff
@@ -1,6 +1,6 @@
-@import url("bootstrap.css") layer(framework);
+@import url('bootstrap.css') layer(framework);
 
 @layer overrides {
```
The bottom is very tight to the edge of the screen. Maybe add a bit of space on html or body?

stacyk replied:
fixed

mirisuzanne reviewed (approved):
looks great

jgerigmeyer committed:
do not load syntax highlighting for ie

jgerigmeyer committed:
Only add style if one was removed.

jgerigmeyer committed:
Drop support for IE

stacyk committed:
remove unused

stacyk committed:
adjust space, move sample content, add layers info

stacyk committed:
Merge branch 'layer-style' of https://github.com/oddbird/cascade-layers-polyfill into layer-style

stacyk committed:
remove 1 sample button

stacyk committed:
add grid to primary to prevent code from making primary too wide

stacyk commented:
@jgerigmeyer @mirisuzanne I edited the text for the sample content to explain a bit more why they are there but maybe they aren't needed since I am not showing how layers work, but rather it is not visually showing anything here in the demo. The code shows how to import bootstrap in a low level layer, but I don't know if anyone would clone this demo to see how it works? 

In any case, I moved this sample content over to the right below the form. I also fixed the code from pushing the primary column wider on smaller screens. Ready for your final reviews.
